src/overlays/actors/ovl_Obj_Timeblock/z_obj_timeblock.c

```diff
@@ -27,41 +57,314 @@ const ActorInit Obj_Timeblock_InitVars = {
+
+static ObjTimeblockSizeOptions sSizeOptions[] = {
+    { 1.0, 60.0, 0x0018 },
+    { 0.60, 40.0, 0x0019 },
+};
+
+static f32 sRanges[] = { 60.0, 100.0, 140.0, 180.0, 220.0, 260.0, 300.0, 300.0 };
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_U8(unk_1F, 2, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1800, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 300, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1500, ICHAIN_STOP),
+};
+
+static Color_RGB8 sPrimColors[] = { { 100, 120, 140 }, { 80, 140, 200 }, { 100, 150, 200 }, { 100, 200, 240 },
+                                    { 80, 110, 140 },  { 70, 160, 225 }, { 80, 100, 130 },  { 100, 110, 190 } };
+
+u32 ObjTimeblock_CalculateIsVisible(ObjTimeblock* this) {
+    if (!((this->dyna.actor.params >> 10) & 1)) {
+        u32 save = this->save;
+        if (save == OBJ_TIMEBLOCK_DO_NOT_SAVE) {
+            return this->unk_175;
+        } else {
+            u8 defaultVisibility = BOOL((this->dyna.actor.params >> 15) & 1);
+            if (save == OBJ_TIMEBLOCK_SAVE) {
+                return this->unk_174 ^ defaultVisibility;
+            } else {
+                u8 linkIsYoung = BOOL(LINK_AGE_IN_YEARS == YEARS_CHILD);
+                return this->unk_174 ^ defaultVisibility ^ linkIsYoung;
+            }
+        }
+    } else {
+        return BOOL((this->dyna.actor.params >> 15) & 1) ^ this->unk_174;
+    }
+}
+
+void ObjTimeblock_SpawnDemoEffect(ObjTimeblock* this, GlobalContext* globalCtx) {
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_DEMO_EFFECT, this->dyna.actor.posRot.pos.x,
+                this->dyna.actor.posRot.pos.y, this->dyna.actor.posRot.pos.z, 0, 0, 0,
+                sSizeOptions[(this->dyna.actor.params >> 8) & 1].demoEffectParams);
+}
+
+void ObjTimeblock_ToggleSwitchFlag(GlobalContext* globalCtx, s32 flag) {
+    if (Flags_GetSwitch(globalCtx, flag)) {
+        Flags_UnsetSwitch(globalCtx, flag);
+    } else {
+        Flags_SetSwitch(globalCtx, flag);
+    }
+}
+
+void ObjTimeblock_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjTimeblock* this = THIS;
+    s32 pad;
+    UNK_PTR colHeader = NULL;
+
+    DynaPolyInfo_SetActorMove(&this->dyna, DPM_UNK);
+    this->dyna.actor.shape.rot.z = 0;
+    this->dyna.actor.posRot.rot.z = this->dyna.actor.shape.rot.z;
+
+    DynaPolyInfo_Alloc(&D_06000B30, &colHeader);
+
+    this->dyna.dynaPolyId =
+        DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    Actor_SetScale(&this->dyna.actor, sSizeOptions[(this->dyna.actor.params >> 8) & 1].scale);
+
+    if ((this->dyna.actor.params >> 6) & 1) {
+        this->save = OBJ_TIMEBLOCK_DO_NOT_SAVE;
+    } else {
+        this->save = ((this->dyna.actor.params & 0x3F) < 0x38) ? OBJ_TIMEBLOCK_SAVE_AGE_INVERTED : OBJ_TIMEBLOCK_SAVE;
+    }
+
+    this->songObserverFunc = &ObjTimeblock_WaitForOcarina;
```

Comment:
```suggestion
    this->songObserverFunc = ObjTimeblock_WaitForOcarina;
```
don't need a `&` for function pointers (also in other places in the file)